src/main/java/bio/terra/service/tabulardata/google/BigQueryProjectProvider.java

```diff
@@ -0,0 +1,21 @@
+package bio.terra.service.tabulardata.google;
+
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * Given a projectId, returns a {@link BigQueryProject} object while caching
+ */
+@Component
+public class BigQueryProjectProvider implements Function<String, BigQueryProject> {
```

Comment:
It seems odd to me to have a injectable component for a static hash map, if the point of making it injectable is so you can mock it for testing. If you want to have a specific project returned for a projectId, how about adding a `put(BigQueryProject)` method to the class? You could add a warning to only use it for testing, although even if it was used in a non-test situation I don't see what the harm would be.

If you did that, then you could replace uses like `new BigQueryProjectProvider().apply()`  with `BigQueryProjectProvider.get()` (or whatever it should be called).

Reply:
Actually, I might turn that around.  I don't love that it uses a static hashmap. I may keep it as a component but convert the cache into a class variable

Reply:
Why not use a static hashmap? It's a memory leak, sure, but that's how it was written before. Maybe create a ticket to look at converting it to a cache instead of a hashmap?

Either way I don't see why it's necessary to make it injectable for this PR. Adding `put()` would let you write the tests.